### ISSUE TITLE
Discussion for creating the ArrayFire .net API 

### ISSUE THREAD

**Issue by pavanky** [arrayfire/arrayfire-dotnet#1]

**Comment by pavanky** [comment]
@umar456 @9prady9 We do not need to do it right away, but if you guys have ideas, this would be the place to discuss.

**Comment by 9prady9** [comment]
We can start with C++/CLI to facilitate usage from other languages in .NET family.

**Comment by royalstream** [comment]
Hi,

As per the conversation I had with Pavan Yalamanchili I forked and created a draft of a wrapper and just did a PR
(see https://groups.google.com/forum/#!topic/arrayfire-users/1N0qYpZFC-Q)

All the boilerplate Interop code is automatically generated from your header (*.h) files which means newer functions should be easy to add. I included the source for the F# tool that performs this task.

The rest of the code is just a shell around the interop static methods.

Following Pavan's advice, the garbage collector is called following a similar behavior to your R wrapper.

I include two example projects using the library, one in C# and one in F#
